website/content/docs/upgrading/plugins.mdx

```diff
@@ -6,206 +6,184 @@ description: These are general upgrade instructions for Vault plugins.
 
 # Upgrading Vault Plugins
 
-## External Plugin Upgrade Procedure
+## Plugin Upgrade Procedure
 
-The following procedure details steps for upgrading an external plugin that has
-been registered to the catalog on a running server. This procedure is
-applicable to secret engines, auth methods, and database plugins.
+The following procedures detail steps for upgrading a plugin that has been mounted
+at a path on a running server. The steps are the same whether the plugin being
+upgraded is built-in or external.
 
-Vault executes plugin binaries when they are configured and roles are established
-around them. The binary cannot be modified or replaced while running, so
-upgrades cannot be performed by simply swapping the binary and updating the hash
-in the plugin catalog.
+~> Plugin versioning was introduced with Vault 1.12.0, so if your Vault server is
+  on 1.11.x or earlier, see the [1.11.x version of this page](/docs/v1.11.x/upgrading/plugins)
+  for plugin upgrade instructions.
 
-Instead, you can restart or reload a plugin with the
-`sys/plugins/reload/backend` [API][plugin_reload_api]. Follow these steps to
-replace or upgrade a Vault plugin binary:
+### Upgrading auth and secrets plugins
 
-1. [Register][plugin_registration] version 1 of `my-db-plugin` to the catalog.
-   Skip this step if your plugin is already registered.
+The process is nearly identical for auth and secret plugins. If you are upgrading
+an auth plugin, just replace all usages of `secrets` or `secret` with `auth`.
+
+1. [Register][plugin_registration] the first version of your plugin to the catalog.
+   Skip this step if your initial plugin is built-in or already registered.
 
     ```shell-session
-    $ vault plugin register -sha256=<SHA256 Hex value of the plugin binary> \
-        database \                  # type
-        my-db-plugin
+    $ vault plugin register
+        -sha256=<SHA256 Hex value of the plugin binary> \
+        secret \
+        my-secret-plugin
     ```
 
-2. [Mount][plugin_management] the plugin backend. Skip this step if the backend
+2. [Mount][plugin_management] the plugin. Skip this step if your initial plugin
```

Comment:
I think we're supposed to use `1.` for each step and let autonumbering keep them consistent.